sauce/features/budget/budget-balance-to-zero/index.js

```diff
@@ -0,0 +1,166 @@
+import { Feature } from 'core/feature';
+import * as toolkitHelper from 'helpers/toolkit';
+
+export class BudgetBalanceToZero extends Feature {
+  attachedObserver = false
+  budgetView = null
+
+  shouldInvoke = () => {
+    return toolkitHelper.getCurrentRouteName().indexOf('budget') !== -1;
+  }
+
+  invoke = () => {
+    if (!this.attachedObserver) {
+      this.addBudgetVersionIdObserver();
+      this.attachedObserver = true;
+    }
+
+    const categories = this.getCategories();
+    const categoryName = this.getInspectorName();
+    const masterCategoryViewId = $('ul.is-checked').prevAll('ul.is-master-category').attr('id');
+
+    if (masterCategoryViewId) {
+      const masterCategory = ynabToolKit.shared.getEmberView(masterCategoryViewId).get('data');
+      const masterCategoryId = masterCategory.get('categoryId');
+
+      categories.forEach(category => {
+        if (category.name === categoryName && category.masterCategoryId === masterCategoryId) {
+          this.updateInspectorButton(category);
+
+          return false;
+        }
+      });
+    }
+  }
+
+  observe = (changedNodes) => {
+    if (this.shouldInvoke() &&
+      (
+        changedNodes.has('inspector-quick-budget') ||
+        changedNodes.has('budget-inspector-default')
+      )) {
+      this.invoke();
+    }
+  }
+
+  addBudgetVersionIdObserver = () => {
+    const resetBudgetView = () => {
```

Comment:
What about using the onRouteChanged() function from the Feature base class?

Reply:
Yup, could certainly do that, this was just the code that was there. I'll have a look now.